src/amo/components/Addon/index.js

```diff
@@ -65,17 +68,33 @@ export class AddonBase extends React.Component {
   }
 
   componentWillMount() {
-    const { addon, dispatch, errorHandler, params } = this.props;
+    const {
+      addon,
+      dispatch,
+      errorHandler,
+      params,
+    } = this.props;
 
     if (addon) {
       dispatch(setViewContext(addon.type));
+      dispatch(fetchOtherAddonsByAuthors({
+        addonType: addon.type,
```

Comment:
It's kind of a bummer that you have to wait for `addon.type` here because it means the two add-on fetching API requests can't happen in parallel. This also means you could have gone with `byAddonId` in the state instead, which is a bit more stable. However, that's probably not worth changing.